docs/pics/publish.dot

```diff
@@ -18,7 +18,7 @@ digraph G {
 
     subgraph cluster_publish {
         label="Publish with audb"
-        pack[label="Pack"]
+        pack[label="(Pack)"]
```

Comment:
I understand that  `()`  flags that a given step is optional. 
My understanding is that there are differnt optionalities:

- induced by the presence of a a precursor version (deps)
-  induced by using parquet that (packing, unpacking)
- induced by the flavor that is requested (conversion)

Would it make sense to distinguish between these?
Not sure whether this would be a good idea or too complicated, but one could  find a notation that distinguishes between the kind of optionalites by in the dot figure by using `[],(),{}`   for each of them in turn.
If too compllicated one could also detail this in the enumeration below. 

I also wonder about the `(Conv.)Media` in the cache: Here `()` denotes converted and not optionality.

Reply:
You are right about the different types of reasons for why something is optional.
I personally prefer if the documentation is not too complicated to read and would stay with hiding the different reasons from the reader.

The reasons might be interested for a developer. Maybe we should think about at some point to at a dedicated section to the documentation, or extend the documentation (and structure) inside the single code files. I would prefer the later, as I have to read the files anyway as a developer.

> I also wonder about the (Conv.)Media in the cache: Here () denotes converted and not optionality.

The conversion is optional. If you just call `audb.load(name)` it will download the data as they are stored on the server and no conversion is applied.

Reply:
>> The conversion is optional. If you just call audb.load(name) it will download the data as they are stored on the server and no conversion is applied.

True.

>> You are right about the different types of reasons for why something is optional.
I personally prefer if the documentation is not too complicated to read and would stay with hiding the different reasons from the reader.

>> The reasons might be interested for a developer. Maybe we should think about at some point to at a dedicated section to the documentation, or extend the documentation (and structure) inside the single code files. I would prefer the later, as I have to read the files anyway as a developer.


This also can be handled in the follow up #439 that you mention above.
Therefore resolving.  I tried to play a little bit with generating call graphs but this is hard for both `load` and `publish`, and reading the code is still the best way to get a grasp. 

Resolving this here.

Reply:
It's even further complicated by the fact that `load` and `load_to` do not behave the same, e.g. in `load_to` we don't copy from the cache of other versions at the moment.